Yesod/Auth/OAuth2.hs

```diff
@@ -104,5 +108,21 @@ authOAuth2 name oauth getCreds = AuthPlugin name dispatch login
         <a href=@{tm $ oauth2Url name}>Login via #{name}
         |]
 
+-- | Handle the common case of fetching Profile information a JSON endpoint
+--
+-- Throws @'InvalidProfileResponse'@ if JSON parsing fails
+--
+handleProfileResponse :: FromJSON a
```

Comment:
I'm not used to worrying about naming so much in Haskell. Can I just call it `f`?

How about `fromProfileURL`?